subsys/bluetooth/controller/Kconfig

```
@@ -121,38 +121,39 @@ config BT_CTLR_SUBRATING_SUPPORT
 config BT_CTLR_CHANNEL_SOUNDING_SUPPORT
 	bool
 
-config BT_CTLR
-	bool "Bluetooth Controller"
-	help
-	  Enables support for SoC native controller implementations.
-
-if BT_CTLR
+# Virtual option that all local LL implemenations should select
+config HAS_BT_CTLR
+	bool
 
-choice BT_LL_CHOICE
-	prompt "Bluetooth Link Layer Selection"
+config BT_CTLR
+	bool "Bluetooth Controller [DEPRECATED]"
+	select HAS_BT_CTLR
+	select DEPRECATED
 	help
-	  Select the Bluetooth Link Layer to compile.
-	  The link layer choice list can be extended out of tree.
-	  In the nRF Connect SDK, the SoftDevice Controller is selected
-	  and set as the default Bluetooth Controller.
+	  Enables support for SoC native controller implementations. This option
+	  has been deprecated and should not be used anymore. Instead, the
+	  Kconfig options for specific Link Layer implementations should select
+	  the HAS_BT_CTLR option.
 
 config BT_LL_SW_SPLIT
 	bool "Software-based BLE Link Layer"
+	default y
 	depends on DT_HAS_ZEPHYR_BT_HCI_LL_SW_SPLIT_ENABLED
```

To make integration simpler with the SoftDevice Controller, I believe we should change this to:
```suggestion
	default y if DT_HAS_ZEPHYR_BT_HCI_LL_SW_SPLIT_ENABLED
```

---

That doesn't seem right to me. Not having the explicit dependency would mean that you could enable the option even though the corresponding DT node isn't enabled, which isn't correct. The following pattern is what almost all drivers in Zephyr use in their Kconfig, so I think it makes sense to follow the same practice:
```
default y
depends on DT_HAS_<compat>_ENABLED
```